flake.nix

```diff
@@ -0,0 +1,85 @@
+{
+  description = "yesod-static-remote";
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs?ref=nixos-22.05";
+    flake-utils.url = "github:numtide/flake-utils";
+    pre-commit-hooks.url = "github:cachix/pre-commit-hooks.nix";
+  };
+
+  outputs =
+    { self
+    , nixpkgs
+    , flake-utils
+    , pre-commit-hooks
+    }:
+    flake-utils.lib.eachSystem [ "x86_64-linux" ] (system:
+    let
+      pkgs = import nixpkgs {
+        inherit system;
+        overlays = [
+          self.overlays.${system}
+        ];
+      };
+    in
+    {
+      overlays = final: prev:
+        with final.haskell.lib;
+        {
+          haskellPackages =
+            prev.haskellPackages.override (
+              old:
+              {
+                overrides =
+                  final.lib.composeExtensions (old.overrides or (_: _: { })) (
+                    self: super: {
+                      yesod-static-remote = dontCheck (failOnAllWarnings (self.callPackage ./default.nix { }));
```

Comment:
You can get rid of `./default.nix` and instead use the function `callCabal2Nix` directly. See https://tek.brick.do/how-to-override-dependency-versions-when-building-a-haskell-project-with-nix-K3VXJd8mEKO7

Reply:
@srid I specifically tried not doing it that way to avoid IFD.
What's your opinion on that?

Reply:
The convenience of not having to remember to regenerate the default.nix upon every change to the cabal file overweighs having to avoid IFD for me. I also like to have smaller Nix that is not spread across multiple files as much as possible.

IFD has so far impacted one place: `nix flake check`, but I workaround that using https://github.com/srid/check-flake

Reply:
> The convenience of not having to remember to regenerate the default.nix upon every change to the cabal file overweighs having to avoid IFD for me. 

I agree, that's why I've automated it.